omod/src/main/java/org/openmrs/module/radiology/web/controller/RadiologyOrderFormController.java

```
@@ -340,34 +340,89 @@ private void populate(ModelAndView mav, Order order, Study study) {
 		return performedStatuses;
 	}
 	
+	/**
+	 * Return true if the current user is authenticated as a referring physician
+	 * 
+	 * @return true if the current user is authenticated as a referring physician
+	 * @should return true if the current user is authenticated as a referring physician
+	 * @should return false if the current user is not authenticated as a referring physician
+	 * @should return false if the current user is not authenticated
+	 */
 	@ModelAttribute("isUserReferringPhysician")
 	private boolean isUserReferringPhysician() {
 		
-		return Context.getAuthenticatedUser().hasRole(REFERRRING_PHYSICIAN, true);
+		if (Context.isAuthenticated()) {
```

I don't think this is a good approach because the API user will not get a definitive answer if the result is false because its a different role or because the user is not authenticated. I would throw an exception for not authenticated and true/false for the role